### PR TITLE
Adding board files for Omnibus F4 NanoV6 (only V6)...bench tested all ports/functions

### PR DESCRIPTION
**Pull request by Hwurzburg**
Only thing I couldn't test was compass since I don't have one with pull-ups required on I2C bus for this device
Bootloader tested also

### CODE REVIEW

File: libraries/AP_HAL_ChibiOS/hwdef/omnibusNanoV6/hwdef.dat [hunk 1]
```diff
@@ -0,0 +1,143 @@
...
+PC8 SBUS_INVERT_RX OUTPUT LOW
+PC9 SBUS_INVERT_TX OUTPUT LOW
+
+# SPI Device table 
+SPIDEV mpu6000    SPI1 DEVID1 MPU6000_CS MODE3 1*MHZ 8*MHZ
+SPIDEV sdcard     SPI2 DEVID2 SDCARD_CS MODE0 400*KHZ 25*MHZ
+SPIDEV bmp280     SPI3 DEVID3 BMP280_CS MODE3 1*MHZ 8*MHZ
+SPIDEV osd        SPI3 DEVID4 OSD_CS MODE0 10*MHZ 10*MHZ
+
+define CONFIG_HAL_BOARD_SUBTYPE HAL_BOARD_SUBTYPE_CHIBIOS_OMNIBUSF4PRO
```
[the remark this answers is not in the document]
Comment (Hwurzburg) [reply libraries/AP_HAL_ChibiOS/hwdef/omnibusNanoV6/hwdef.dat]: do I just create a name...do I have to add it anywhere else?

[the remark this answers is not in the document]
Comment (Hwurzburg) [reply libraries/AP_HAL_ChibiOS/hwdef/omnibusNanoV6/hwdef.dat]: changed name

File: libraries/AP_HAL_ChibiOS/hwdef/omnibusNanoV6/hwdef.dat [hunk 2]
```diff
@@ -0,0 +1,143 @@
...
+# external compass types
+define ALLOW_ARM_NO_COMPASS
+define HAL_COMPASS_DEFAULT HAL_COMPASS_NONE
+define HAL_PROBE_EXTERNAL_I2C_COMPASSES
+define HAL_I2C_INTERNAL_MASK 0
+
+define STORAGE_FLASH_PAGE 1
+define HAL_STORAGE_SIZE 15360
+
+define HAL_OS_FATFS_IO 1
```
[the remark this answers is not in the document]
Comment (Hwurzburg) [reply libraries/AP_HAL_ChibiOS/hwdef/omnibusNanoV6/hwdef.dat]: removed

File: libraries/AP_HAL_ChibiOS/hwdef/omnibusNanoV6/hwdef.dat [hunk 3]
```diff
@@ -0,0 +1,143 @@
+# hw definition file for processing by chibios_pins.py
+# Omnibus F4 Nano V6 only
+# with F405 mcu, mpu6000 imu, bmp280 barometer, 7456 series osd  and sdcard
```
[the remark this answers is not in the document]
Comment (Hwurzburg) [reply libraries/AP_HAL_ChibiOS/hwdef/omnibusNanoV6/hwdef.dat]: revised

File: libraries/AP_HAL_ChibiOS/hwdef/omnibusNanoV6/hwdef.dat (hunk 1 above)
[the remark this answers is not in the document]
Comment (sh83) [reply libraries/AP_HAL_ChibiOS/hwdef/omnibusNanoV6/hwdef.dat]: Yes, add it here https://github.com/ArduPilot/ardupilot/blob/master/libraries/AP_HAL/AP_HAL_Boards.h

File: libraries/AP_HAL_ChibiOS/hwdef/omnibusNanoV6/hwdef.dat [hunk 4]
```diff
@@ -0,0 +1,136 @@
...
+PB6 TIM4_CH1 TIM4 RCININT PULLDOWN LOW
+
+#Omnibus F4 V3 and later had hw inverter on UART6
+#Overide it to use as GPS UART port
+PC8 SBUS_INVERT_RX OUTPUT LOW
+PC9 SBUS_INVERT_TX OUTPUT LOW
+
+# SPI Device table 
+SPIDEV mpu6000    SPI1 DEVID1 MPU6000_CS MODE3 1*MHZ 8*MHZ
+SPIDEV sdcard     SPI2 DEVID2 SDCARD_CS MODE0 400*KHZ 25*MHZ
```
Comment (sh83) [inline libraries/AP_HAL_ChibiOS/hwdef/omnibusNanoV6/hwdef.dat]: Sorry, and here

File: libraries/AP_HAL_ChibiOS/hwdef/omnibusNanoV6/hwdef.dat [hunk 5]
```diff
@@ -0,0 +1,136 @@
...
+PA7 SPI1_MOSI SPI1
+
+# note that this board needs PULLUP on I2C pins
+PB10 I2C2_SCL I2C2 PULLUP
+PB11 I2C2_SDA I2C2 PULLUP
+
+PB15 SPI2_MOSI SPI2
+PB14 SPI2_MISO SPI2
+PB13 SPI2_SCK SPI2
+PB12 SDCARD_CS CS
```
Comment (sh83) [inline libraries/AP_HAL_ChibiOS/hwdef/omnibusNanoV6/hwdef.dat]: No sdcard ...

File: libraries/AP_HAL_ChibiOS/hwdef/omnibusNanoV6/hwdef.dat [hunk 6]
```diff
@@ -0,0 +1,136 @@
...
+
+PA11 OTG_FS_DM OTG1
+PA12 OTG_FS_DP OTG1
+PC5 VBUS INPUT OPENDRAIN
+
+#LED strip output pad used for RC input
+PB6 TIM4_CH1 TIM4 RCININT PULLDOWN LOW
+
+#Omnibus F4 V3 and later had hw inverter on UART6
+#Overide it to use as GPS UART port
```
Comment (sh83) [inline libraries/AP_HAL_ChibiOS/hwdef/omnibusNanoV6/hwdef.dat]: I think this is not needed, cause no controlled inverter here

**Comment by sh83** [comment]
Looks good to me, thank you!

File: libraries/AP_HAL_ChibiOS/hwdef/omnibusNanoV6/hwdef.dat (hunk 1 above)
[the remark this answers is not in the document]
Comment (tridge) [reply libraries/AP_HAL_ChibiOS/hwdef/omnibusNanoV6/hwdef.dat]: you don't actually need a board subtype, unless the board has special build requirements. Most new boards don't need it

**Comment by tridge** [comment]
I've done a few cleanups and rebase

**Comment by Davidsastresas** [comment]
> Only thing I couldn't test was compass since I don't have one with pull-ups required on I2C bus for this device
> Bootloader tested also

Did you had a chance to test any suitable compass? I am playing around with this board on a micro helicopter and I can not manage to make the compass nor the gps work. I didn't know about the pullups needed on the i2c port, that could be the problem with the compass, although the orientation is reporting all right in ground station, it never succesfully calibrate. About the GPS it is connected to serial 6 and the protocols set all right, and the scope shows the packets being sent from GPS. It is a m8q from mrobotics.

Everything else is working really nice!

Thank you.

Regards.

**Comment by Hwurzburg** [comment]
yes, I finally added some pullups and tested with a UBLOX M6N gps with HMC5883 compass combo board....worked fine...
many times I find the "on-board" cal does not complete for some reason (maybe I am not performing it correctly)...I use an older MP version with "live" calibration and it always seems to work...

**Pull request merged**